### PR TITLE
Add LZ4 1.7.5 recipe.

### PR DESCRIPTION
**Pull request by rolando**
Fixes https://github.com/conda-forge/staged-recipes/issues/1906

### CODE REVIEW

**Comment by rolando** [comment]
Windows x64 is failing with:
```
     5>datagen.obj : error LNK2001: unresolved external symbol __imp___acrt_iob_func [C:\bld\lz4-c_1487478235468\work\lz4-1.7.5\visual\VS2010\lz4\lz4.vcxproj]
     5>bench.obj : error LNK2001: unresolved external symbol __imp___stdio_common_vfprintf [C:\bld\lz4-c_1487478235468\work\lz4-1.7.5\visual\VS2010\lz4\lz4.vcxproj]
     5>bench.obj : error LNK2001: unresolved external symbol __imp___stdio_common_vsprintf [C:\bld\lz4-c_1487478235468\work\lz4-1.7.5\visual\VS2010\lz4\lz4.vcxproj]
```

I tried some fixes like trying to link `legacy_stdio_definitions.lib` but it wasn't successful. I've followed as close as possible the build instructions from LZ4 itself, which has successful win builds: https://ci.appveyor.com/project/YannCollet/lz4-1lndh

**Comment by jakirkham** [comment]
ref: http://stackoverflow.com/a/30894349 ?

**Comment by rolando** [comment]
@jakirkham I tried adding `/p:AdditionalDependencies=legacy_stdio_definitions.lib` but didn't work. Then I tried patching the project file and got
```
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\x64_Release\lz4.exe" /INCREMENTAL:NO /NOLOGO setargv.obj legacy_stdio_definitions.lib kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\x64_Release\lz4.pdb" /SUBSYSTEM:CONSOLE /OPT:REF /OPT:ICF /LTCG:incremental /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\x64_Release\lz4.lib" /MACHINE:X64 "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4.res"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4frame.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4hc.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\xxhash.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\datagen.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\bench.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4cli.obj"
         "C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\bin\obj\lz4_x64_Release\lz4io.obj"
     5>LINK : fatal error LNK1181: cannot open input file 'legacy_stdio_definitions.lib' [C:\bld\lz4-c_1487525177003\work\lz4-1.7.5\visual\VS2010\lz4\lz4.vcxproj]
```

**Comment by jakirkham** [comment]
Maybe we can just skip Windows for now and investigate it on the feedstock.

**Comment by rolando** [comment]
@jakirkham sounds good to me. What puzzles me is that lz4 has windows builds what's different in our script https://ci.appveyor.com/project/YannCollet/lz4-1lndh

Also we are able to build the package for `win-32+vc14` but not for `win-64+vc14`: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.11633/job/tyaoi1l3g99ioc3u

File: recipes/lz4-c/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/{{ author }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256sum }}
+  patches:
+   - patches/0001-Do-not-build-tests-binaries-on-VS-project.patch  # [win]
+
+build:
+  number: 0
+  skip: True  # [win and py2k]
```
Comment (jakirkham) [inline recipes/lz4-c/meta.yaml]: Would change to `win and not py35`.

File: recipes/lz4-c/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  features:
+    - vc14  # [win and py>=35]
+
+requirements:
+  build:
+    - toolchain
+    - python     # [win]
+    - vc 14      # [win and py>=35]
```
Comment (jakirkham) [inline recipes/lz4-c/meta.yaml]: Would switch to `py35`. `conda-smithy` renders things incorrectly if they have comparisons in the selectors on dependencies.

xref: https://github.com/conda-forge/conda-smithy/issues/257

**Comment by jakirkham** [comment]
Maybe `legacy_stdio_definitions.lib` is 32-bit and not 64-bit?

**Comment by jakirkham** [comment]
> If your project links with static libraries that were compiled with a release of Visual C++ earlier than 2015, the linker might report an unresolved external symbol. These errors might reference internal stdio definitions for _iob, _iob_func, or related imports for certain stdio functions in the form of \_imp\_*. Microsoft recommends that you recompile all static libraries with the latest version of the Visual C++ compiler and libraries when you upgrade a project. If the library is a third-party library for which source is not available, you should either request an updated binary from the third party or encapsulate your usage of that library into a separate DLL that you compile with the older version of the Visual C++ compiler and libraries.

ref: https://msdn.microsoft.com/en-us/library/bb531344.aspx

File: recipes/lz4-c/bld.bat [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    set ADDITIONALPARAM=/p:LibraryPath="C:\Program Files\Microsoft SDKs\Windows\v7.1\lib\x64;c:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\lib\amd64;C:\Program Files (x86)\Microsoft Visual Studio 10.0\;C:\Program Files (x86)\Microsoft Visual Studio 10.0\lib\amd64;"
+)
+set CONFIGURATION=Release
+set VSPROJ_DIR=%SRC_DIR%\visual\VS2010
+set BUILD_DIR=%VSPROJ_DIR%\bin\%PLATFORM%_%CONFIGURATION%
+msbuild.exe /m ^
+    /p:PlatformToolset=v140 ^
+    /p:Platform=%PLATFORM% ^
+    /p:Configuration=%CONFIGURATION% %ADDITIONALPARAM% ^
+    /p:AdditionalDependencies=legacy_stdio_definitions.lib ^
```
Comment (jakirkham) [inline recipes/lz4-c/bld.bat]: Maybe we can try adding `legacy_stdio_wide_specifiers.lib` too?

**Comment by rolando** [comment]
Found out that we don't need the custom library path, that was messing up our build.

**Comment by jakirkham** [comment]
Nice work on fixing Windows, @rolando!

Still waiting for the Travis CI [build]( https://travis-ci.org/conda-forge/staged-recipes/builds/204094447 ) even though it's status does not show up here. FYI have raised this issue with Travis CI.

**Comment by rolando** [comment]
:tada:

**Comment by rolando** [comment]
Travis build failed with
```
Building all recipes.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5215  100  5215    0     0  37591      0 --:--:-- --:--:-- --:--:-- 37789
Checking to see if this PR build is outdated.
Traceback (most recent call last):
  File "<stdin>", line 192, in <module>
  File "<stdin>", line 181, in main
  File "<stdin>", line 93, in travis_check_latest_pr_build
ValueError: max() arg is an empty sequence
```

**Comment by jakirkham** [comment]
Interesting. That looks like an issue with the fast finish script. Can you please report it [here]( https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues )?

**Comment by jakirkham** [comment]
Seems this now works with the fast finish script fix in PR ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/59 ).

**Pull request merged**